geomesa-index-api/src/main/scala/org/locationtech/geomesa/index/planning/LocalQueryRunner.scala

```diff
@@ -237,7 +237,11 @@ object LocalQueryRunner {
     } else if (hints.isDensityQuery) {
       val Some(envelope) = hints.getDensityEnvelope
       val Some((width, height)) = hints.getDensityBounds
-      densityTransform(sampled, sft, envelope, width, height, hints.getDensityWeight)
+      val geom = hints.getDensityGeometry.getOrElse(sft.getGeometryDescriptor.getLocalName)
+      require(
```

Comment:
Meh, since this require is in a few places, it is worth having a method that gets the DensityGeometry and throws an exception if it cannot?

Reply:
yeah, i didn't want to make the RichHints too complicated but i can add a method in DensityScan.